File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        int n=nums.size();
        vector<int> freq(101,0);
        for(int x: nums){
            freq[x]++;
        }
        for(int i=1;i<=100;i++){
            freq[i]+=freq[i-1];
        }
        vector<int> result;
        for(int x:nums){
            if(x==0){
                result.push_back(0);
            }else{
                result.push_back(freq[x-1]);
            }
        }
        return result;
    }
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int n = nums.length;
        int[] freq = new int[101]; 
        for(var num : nums){
            freq[num]++;
        }
        for(int i=1;i<=100;i++){
            freq[i]+=freq[i-1];
        }
        int[] res = new int[n];
        for(int i=0;i<n;i++){
            if(nums[i]==0){
                res[i]=0;
            }else{
                res[i]=freq[nums[i]-1];
            }
        }
        return res;
    }
}